instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/ParentContextExtractor.java

```diff
@@ -1,81 +0,0 @@
-/*
- * Copyright The OpenTelemetry Authors
- * SPDX-License-Identifier: Apache-2.0
- */
-
-package io.opentelemetry.instrumentation.awslambdacore.v1_0.internal;
-
-import static io.opentelemetry.instrumentation.awslambdacore.v1_0.internal.MapUtils.lowercaseMap;
-
-import io.opentelemetry.api.trace.Span;
-import io.opentelemetry.api.trace.SpanContext;
-import io.opentelemetry.context.Context;
-import io.opentelemetry.context.propagation.TextMapGetter;
-import io.opentelemetry.contrib.awsxray.propagator.AwsXrayPropagator;
-import java.util.Collections;
-import java.util.Locale;
-import java.util.Map;
-
-/**
- * This class is internal and is hence not for public use. Its APIs are unstable and can change at
- * any time.
- */
-public final class ParentContextExtractor {
-
-  private static final String AWS_TRACE_HEADER_ENV_KEY = "_X_AMZN_TRACE_ID";
-
-  static Context extract(Map<String, String> headers, AwsLambdaFunctionInstrumenter instrumenter) {
-    Context parentContext = null;
-    String parentTraceHeader = System.getenv(AWS_TRACE_HEADER_ENV_KEY);
-    if (parentTraceHeader != null) {
-      parentContext = fromXrayHeader(parentTraceHeader);
-    }
-    if (!isValidAndSampled(parentContext)) {
-      // try http
-      parentContext = fromHttpHeaders(headers, instrumenter);
-    }
```

Comment:
@laurit This is the particularly problematic section in the current releases.  Notice that it first uses the context from an environment variable if available, not the context from the incoming message.  This results in broken traces even if the x-ray context was sent via the aws-sdk as you linked to.  Rather than completely removing the environment variable provided context, this PR instead (and what is now required by the specification) adds that as a span link and restores regular context propagation from the incoming message.

Sorry, I should have explained better in my PR description.